### PR TITLE
Fix x22 cgm page gaps

### PR DESCRIPTION
**Pull request by tmecklem**
Changes to enhance the reliability and correctness of cgm page decoding

Details:

* Change event decoding to add sensor packets, sensor errors, low and high glucose data
* Remove 0x13 and 0x01 events incorrectly marked as offset-incrementing events
* Change so that only sensor timestamp events can serve as reference for relative offset events
* Time offset happens now in reverse chronological order from sensor timestamps (the same direction as decoding happens from end of page to beginning).
* cgm events at the end of the page that are not followed by a sensor timestamp receive their offset from the preceding sensor timestamp, unless the cgm reading event chain is broken by another type of event
* Add glucose history timestamp write command and page re-fetch, that is used when an appropriate reference timestamp is unavailable in the current cgm page.
* Additional attributes of known events are decoded, such as "page_end", "gap" or "last_rf" timestamp type on sensor timestamp.

### CODE REVIEW

File: MinimedKit/GlucoseEvents/SensorCalGlucoseEvent.swift [hunk 1]
```diff
@@ -11,10 +11,10 @@ import Foundation
 public struct SensorCalGlucoseEvent: RelativeTimestampedGlucoseEvent {
     public let length: Int
     public let rawData: Data
-    public let waiting: String
-    public var timestamp: DateComponents
+    public let calibrationType: String
```
Comment (loudnate) [inline MinimedKit/GlucoseEvents/SensorCalGlucoseEvent.swift]: If this value is useful to public clients, its options should be defined as an enum. If it's just for debugging in `dictionaryRepresentation`, then it should be made `private`.

This comment also applies similar approaches below: `errorType`, `packetType`, `statusType`

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 2]
```diff
@@ -474,6 +478,18 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
                     }
                     return NSLocalizedString("Fetching glucose…", comment: "Progress message for fetching pump glucose.")
                 }
+            case .writeGlucoseHistoryTimestamp:
+                vc = CommandResponseViewController { [unowned self] (completionHandler) -> String in
+                    self.device.ops?.writeGlucoseHistoryTimestamp() { (response) -> Void in
+                        switch response {
+                        case .success(_):
+                            completionHandler(String(describing: "Okay, thing worked"))
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: `String(describing: "Okay, thing worked")` is more concisely written as `"Okay, thing worked"`. Future users might thank you if this was more descriptive...

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift [hunk 3]
```diff
@@ -474,6 +478,18 @@ public class RileyLinkDeviceTableViewController: UITableViewController, TextFiel
...
+                        case .failure(let error):
+                            completionHandler(String(describing: error))
+                        }
+                    }
+                    return NSLocalizedString("Fetching glucose…", comment: "Progress message for fetching pump glucose.")
```
Comment (loudnate) [inline RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: This is copied from line 479

File: RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift (hunk 2 above)
Comment (tmecklem) [reply RileyLinkKit/UI/RileyLinkDeviceTableViewController.swift]: LOL, didn't mean to leave that in there.

File: MinimedKit/GlucoseEvents/SensorCalGlucoseEvent.swift (hunk 1 above)
Comment (tmecklem) [reply MinimedKit/GlucoseEvents/SensorCalGlucoseEvent.swift]: So far, the additional information is useful for diagnosing problems but there's no different actions taken based on any of it. I'll change those to private.

**Pull request merged**